### PR TITLE
Verify the signature on dotfiles when creating a homedir

### PR DESCRIPTION
**Pull request by KellerFuchs**
This is done in two steps:

1. make a world-readable, role-based, GnuPG keyring;
2. use it to check the signature of the git `HEAD` when cloning the dotfiles repo.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
Don't merge yet, I forgot some etckeeper stuff.

File: skel/.bash_profile [hunk 1]
```diff
@@ -5,6 +5,17 @@
 # get mail file
 mail=$(cat msg.welcome)
 
+# Prepare GnuPG homedir
+export GNUPGHOME=$(mktemp -d --tmpdir gpg.XXXXXX)
+trap "rm -rf -- '${GNUPGHOME}'" EXIT
+
+cat > "${GNUPGHOME}/gpg.conf" <<EOF
+# Never, ever, ever do this in your personal gpg.conf
+# However, this is sane when you know use an empty GNUPGHOME
```
Comment (daurnimator) [inline skel/.bash_profile]: > sane when you know use an empty

huh?

Comment (KellerFuchs) [inline skel/.bash_profile]: Oops, missing word in the comment.
Thanks

**Comment by KellerFuchs** [comment]
I tested it on my `kellertest` account (adapting the thing to have the keyring in `~/hashbang-admins.gpg`, temporarily).

Both the “good” (valid signature) case and the “unknown signer” case work.
I didn't test the “no signature” case, but `git verify-commit` will also error-out on it.

**Comment by KellerFuchs** [comment]
@daurnimator Any comment, now that I fixed the missing word?

**Comment by daurnimator** [comment]
Can't see anything wrong with it. 

Not something I see the point of; but whatever, merge it.

**Comment by KellerFuchs** [comment]
OK.

@daurnimator FYI, the point of hashbang/hashbang#14 is not to trust Github ultimately.
Of course, there is no point in taking care of dotfiles when shell-etc is also pulled from Github, I was just starting with the low-hanging fruit.

**Pull request merged**